src/code/z_play.c

```diff
@@ -354,13 +353,13 @@ void Play_Init(GameState* thisx) {
         !IS_CUTSCENE_LAYER) {
         if (CHECK_QUEST_ITEM(QUEST_KOKIRI_EMERALD) && CHECK_QUEST_ITEM(QUEST_GORON_RUBY) &&
             CHECK_QUEST_ITEM(QUEST_ZORA_SAPPHIRE)) {
-            gSaveContext.sceneLayer = 1;
+            gSaveContext.sceneLayer = SCENE_LAYER_CHILD_NIGHT;
```

Comment:
```suggestion
            gSaveContext.sceneLayer = 1;
```

This is not `SCENE_LAYER_CHILD_NIGHT`, but a special Hyrule Field scene layer for spawning the Ocarina of Time.